kiva/pyagg.py

```diff
@@ -0,0 +1,725 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2016, Enthought, Inc
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in enthought/LICENSE.txt and may be redistributed only
+# under the conditions described in the aforementioned license.  The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+# Thanks for using Enthought open source!
+# -----------------------------------------------------------------------------
+from __future__ import absolute_import, print_function, division
+
+from collections import namedtuple
+import warnings
+
+# Major package imports.
+import numpy as np
+import pyagg as agg
+
+# Local imports.
+from .abstract_graphics_context import AbstractGraphicsContext
+from .fonttools import Font
+import kiva.constants as constants
+
+# These are the symbols that a backend has to define.
+__all__ = ["CompiledPath", "Font", "font_metrics_provider", "GraphicsContext"]
+
+cap_style = {
+    constants.CAP_ROUND: agg.LineCap.CapRound,
+    constants.CAP_SQUARE: agg.LineCap.CapSquare,
+    constants.CAP_BUTT: agg.LineCap.CapButt,
+}
+join_style = {
+    constants.JOIN_ROUND: agg.LineJoin.JoinRound,
+    constants.JOIN_BEVEL: agg.LineJoin.JoinBevel,
+    constants.JOIN_MITER: agg.LineJoin.JoinMiter,
+}
+draw_modes = {
+    constants.FILL: agg.DrawingMode.DrawFill,
+    constants.EOF_FILL: agg.DrawingMode.DrawEofFill,
+    constants.STROKE: agg.DrawingMode.DrawStroke,
+    constants.FILL_STROKE: agg.DrawingMode.DrawFillStroke,
+    constants.EOF_FILL_STROKE: agg.DrawingMode.DrawEofFillStroke,
+}
+gradient_coord_modes = {
+    'userSpaceOnUse': agg.GradientUnits.UserSpace,
+    'objectBoundingBox': agg.GradientUnits.ObjectBoundingBox,
+}
+gradient_spread_modes = {
+    'pad': agg.GradientSpread.SpreadPad,
+    'repeat': agg.GradientSpread.SpreadRepeat,
+    'reflect': agg.GradientSpread.SpreadReflect,
+}
+StateBundle = namedtuple('StateBundle',
+                         ['state', 'path', 'stroke', 'fill', 'transform',
+                          'text_transform', 'font'])
+
+
+class GraphicsContext(object):
```

Comment:
It might make sense to inherit from `AbstractGraphicsContext` or `basecore2d.GraphicsContextBase`.  Although at this point it may be more hassle than its worth.